File: web/camera.js
export class Camera {
    constructor() {
        this.target = "center";
        this.zoom = 1.0;
        this.presets = ["Top", "Side", "Front", "Free"];
    }

    setPreset(preset) {
        if (this.presets.includes(preset)) {
            console.log("Camera preset set to:", preset);
        }
    }
}

File: src/drosophila_pd/flystudio/camera.py
from dataclasses import dataclass, field
from typing import Dict, Any, Tuple

@dataclass
class Camera:
    """Camera configuration and state."""
    id: str
    type: str  # e.g., 'top', 'front', 'side', 'free', 'tracking', 'cinematic'
    position: Tuple[float, float, float] = (0.0, 0.0, 0.0)
    target: Tuple[float, float, float] = (0.0, 0.0, 0.0)
    up: Tuple[float, float, float] = (0.0, 0.0, 1.0)
    properties: Dict[str, Any] = field(default_factory=dict)
